File: src/1961.check-if-string-is-a-prefix-of-array.2093/solution.ts
function isPrefixString (s: string, words: string[]): boolean {
    let index = 0;
    let i = 0;
    let j = 0;
    while (index < s.length && i < words.length) {
        if (s[index] !== words[i][j]) {
            return false;
        }
        index++;
        j++;
        if (j === words[i].length) {
            i++;
            j = 0;
        }
    }
    return index === s.length && j === 0;
}

File: src/1961.check-if-string-is-a-prefix-of-array.2093/solution.java
class Solution {
    public boolean isPrefixString(String s, String[] words) {
        char[] chars = s.toCharArray();
        int index = 0;
        for (String string : words) {
            if(index == chars.length){
                return true;
            }
            if (index + string.length() > chars.length) {
                return false;
            }
            char[] word = string.toCharArray();
            for (char c : word) {
                if (chars[index++] != c) {
                    return false;
                }
            }
        }
        return index == chars.length;
    }
}